main.tf

```diff
@@ -0,0 +1,178 @@
+locals {
+  # Create a result map of all built-in event rules and given custom rules.
+  event_rules = merge(
+    var.enable_ecs_task_state_event_rule ? {
+      ECSTaskStateChange = {
+        detail-type = ["ECS Task State Change"]
+        detail      = var.ecs_task_state_event_rule_detail
+      }
+    } : {},
+    var.enable_ecs_deployment_state_event_rule ? {
+      ECSDeploymentStateChange = {
+        detail-type = ["ECS Deployment State Change"]
+        detail      = var.ecs_deployment_state_event_rule_detail
+      }
+    } : {},
+    var.enable_ecs_service_action_event_rule ? {
+      ECSServiceAction = {
+        detail-type = ["ECS Service Action"]
+        detail      = var.ecs_service_action_event_rule_detail
+      }
+    } : {},
+    var.custom_event_rules
+  )
+}
+
+resource "aws_cloudwatch_event_rule" "this" {
+  for_each = local.event_rules
+
+  name        = "${var.name}-${each.key}"
+  description = "Event rule for ${each.key}"
+  event_pattern = jsonencode({
+    source      = [try(each.value.source, "aws.ecs")]
```

Comment:
This is a dangerous assumption. we should force explicit specification of the source. For example someone can pass ec2 event filter and not specify the source and you will default to ecs source